frontend/src/lib/Quiver.ts

```diff
@@ -443,8 +445,14 @@ export class Quiver {
 
     // Make sure indexes have same types.
     if (!Quiver.sameIndexTypes(this._types.index, otherIndexTypes)) {
-      const receivedIndexTypes = otherIndexTypes.map(i => i.name)
-      const expectedIndexTypes = this._types.index.map(i => i.name)
+      const receivedIndexTypes = otherIndexTypes.map(
+        ({ pandas_type, numpy_type }) =>
+          pandas_type === "object" ? numpy_type : pandas_type
+      )
+      const expectedIndexTypes = this._types.index.map(
+        ({ pandas_type, numpy_type }) =>
+          pandas_type === "object" ? numpy_type : pandas_type
+      )
```

Comment:
Since this logic is used in multiple places, let's pull it into a utility function, something like

```typescript
function getIndexType(type: Type): IndexTypeName | string {
  if (type.pandas_type === "object") { etc etc }
}
```